Sources/StreamChatUI/Utils/UIViewController+Extensions.swift

```diff
@@ -19,4 +19,90 @@ extension UIViewController {
         view.removeFromSuperview()
         removeFromParent()
     }
+
+    var isUIHostingController: Bool {
+        String(describing: self).contains("UIHostingController")
+    }
+
+    /// Helper method to correctly setup navigation of parent.
+    /// Used when view is wrapped using `UIHostingController` and used in SwiftUI
+    /// so all properties of `navigationItem` are propagated correctly.
+    @available(iOS 13.0, *)
+    func setupParentNavigation(parent: UIViewController) {
+        let parentNavItem = parent.navigationItem
+
+        if parentNavItem.backBarButtonItem == nil {
+            parentNavItem.backBarButtonItem = navigationItem.backBarButtonItem
+        }
+
+        if #available(iOS 14.0, *) {
+            if parent.navigationItem.backButtonDisplayMode == .default {
+                parent.navigationItem.backButtonDisplayMode = navigationItem.backButtonDisplayMode
+            }
+        }
+
+        if parentNavItem.backButtonTitle == nil {
+            parentNavItem.backButtonTitle = navigationItem.backButtonTitle
+        }
+
+        if parentNavItem.compactAppearance == nil {
+            parentNavItem.compactAppearance = navigationItem.compactAppearance
+        }
+
+        if parentNavItem.hidesSearchBarWhenScrolling {
+            parent.navigationItem.hidesSearchBarWhenScrolling = navigationItem.hidesSearchBarWhenScrolling
+        }
+
+        if parent.navigationItem.largeTitleDisplayMode == .automatic {
+            parentNavItem.largeTitleDisplayMode = navigationItem.largeTitleDisplayMode
+        }
+
+        if parent.navigationItem.hidesBackButton == false {
+            parent.navigationItem.hidesBackButton = navigationItem.hidesBackButton
+        }
+
+        if let leftBarButtonItems = parentNavItem.leftBarButtonItems {
+            if leftBarButtonItems.isEmpty {
+                parentNavItem.leftBarButtonItems = navigationItem.leftBarButtonItems
+            }
+        } else {
+            parentNavItem.leftBarButtonItems = navigationItem.leftBarButtonItems
+        }
```

Comment:
Maybe this can be simplified as:
```swift
if parentNavItem.leftBarButtonItems == nil || parentNavItem.leftBarButtonItems?.isEmpty == true {
    parentNavItem.leftBarButtonItems = navigationItem.leftBarButtonItems
}
```
We will avoid assigning the value at the places.

Reply:
Sure, why not. Adjusted.